tfjs-core/src/io/io_utils.ts

```diff
@@ -152,16 +154,37 @@ export function decodeWeights(
           new Uint16Array(byteBuffer);
       if (dtype === 'float32') {
         if (quantization.dtype === 'uint8' || quantization.dtype === 'uint16') {
-          values = new Float32Array(quantizedArray.length);
-          for (let i = 0; i < quantizedArray.length; i++) {
-            const v = quantizedArray[i];
-            values[i] = v * quantization.scale + quantization.min;
+          if (env().getBool('FLOAT16') && env().getBool('DRIVER_SUPPORT_FLOAT16'))
```

Comment:
Maybe it's better to put the fp16 checking into a helper function.

Reply:
Let's keep it firstly.